server/src/test/java/io/spine/examples/shareaware/server/e2e/given/EntitySubscription.java

```diff
@@ -68,10 +97,29 @@ private void setState(S value) {
 
         private S state() {
```

Comment:
I am not sure why this method still waits for something. What is the expected scenario? Maybe it should return `@Nullable` state instead?

I also don't understand why `clearState()` still exists, instead of instantiating `future` field each time when we need this "waiting" for the result.

I think, `state()` also deserves some documentation.

Reply:
I still believe that we need to clear the state of the entity before we send a command to the server. I think that the client observer on the entity state is triggered immediately (or after a very little amount of time) after we send a command, so if we gonna clear the state of the entity after sending a command, it will reinitialize the `future` which has been already triggered by the command.

Reply:
@Artem-Semenov-dev the thing here is that `clearState()` does not clear the state of the entity.

I would think that we should remember the last update of `S` value, and return it here. While all the gymnastics with `future`s should stay in `setState` and `waitForUpdate`. 

Let's discuss this vocally.